src/applications/find-forms/containers/SearchResults.jsx

```diff
@@ -23,62 +23,78 @@ import { FAF_SORT_OPTIONS, FAF_TEST_OPTION_CLOSEST_MATCH } from '../constants';
+  useEffect(() => {
+    // This hook is run ONLY after the return statement of the component is evaluated (ComponentDidMount || Update)
+    ref.current = value;
+  });
+  return ref.current; // Returns the value BEFORE the return statement of the component is evaluated.
+};
+/** How it works ^
+ * Essentially it's step by step.
+ * 1. Init of SearchResults which calls usePreviousProps() defining the REF. (Ignores UseEffect because component is not rendered yet)
+ * 2. SearchResults return statement is evaluated.
+ * 3. (CDM) UseEffects is evoked and the variables are assessed. PrevProps is still the same as when it was evoked during step 1.
+ * 4. Consumer fills in the search bar and hits search. That updates Redux which returns NEW Props to component.
+ * 5. (CDU) usePreviousProps(newProps) is invoked; it returns the value of the REF from step 1 is return as it is the "Previous Props".
+ * 6. #2 is repeated.
+ * 7. (CDU) UseEffects is evoked and the variables are assessed. PrevProps is still the same as when it was evoked during step 1. However, the "newProps" from step #5 will now be the new "prevProps" and the REF Value is overwritten.
+ * 8. Repeat
+ */
+
+export const SearchResults = ({
+  error,
+  fetching,
+  page,
+  query,
+  results,
+  sortByPropertyName,
+  hasOnlyRetiredForms,
+  startIndex,
+  useLighthouseSearchAlgo,
+  updatePagination,
+  updateSortByPropertyName,
+}) => {
+  const prevProps = usePreviousProps({
+    fetching,
+    useLighthouseSearchAlgo,
+  });
 
-  componentDidUpdate(previousProps) {
-    const { props } = this;
-    const justRefreshed = previousProps.fetching && !props.fetching;
+  useEffect(() => {
+    const justRefreshed = prevProps?.fetching && !fetching;
     if (justRefreshed) {
       focusElement('[data-forms-focus]');
     }
 
     // NOTE: This is only for testing Lighthouse Search Algorithm
     if (
-      previousProps.useLighthouseSearchAlgo === undefined &&
-      props.useLighthouseSearchAlgo === true
+      prevProps?.useLighthouseSearchAlgo === undefined &&
+      useLighthouseSearchAlgo === true
     ) {
-      props.updateSortByPropertyName(
-        FAF_TEST_OPTION_CLOSEST_MATCH,
-        props.results,
-      );
+      updateSortByPropertyName(FAF_TEST_OPTION_CLOSEST_MATCH, results);
     }
-  }
-
-  onPageSelect = page => {
-    const { results, updatePagination } = this.props;
+  });
 
+  const onPageSelect = p => {
     // Derive the new start index.
-    let startIndex = page * MAX_PAGE_LIST_LENGTH - MAX_PAGE_LIST_LENGTH;
+    let startI = p * MAX_PAGE_LIST_LENGTH - MAX_PAGE_LIST_LENGTH;
 
     // Ensure the start index is not greater than the total amount of results.
-    if (startIndex >= results.length) {
-      startIndex = results.length - 1;
+    if (startI >= results.length) {
+      startI = results.length - 1;
     }
 
     // Update the page and the new start index.
-    updatePagination(page, startIndex);
+    updatePagination(p, startI);
```

Comment:
Could we not abbreviate here and just spell out page and startIndex for clarity? 🙂